### ISSUE TITLE
Role Settings

### ISSUE THREAD

**Issue by dcooney** [dcooney/block-manager#2]
> This plugin could become very powerful if it included the option to configure the block access per user-role.
[source](https://wordpress.org/support/topic/feature-requests-multisite-and-per-role-settings/)

**Comment by CodingSamurai** [comment]
Agreed that this would be an essential feature where we'd want to be able to say certain blocks can only be used by certain roles (for instance only Admins can use the Custom HTML block but Editors, Staff & Admins can use the Gallery block).

**Comment by landwire** [comment]
Are there any plans of implementing this feature request?

**Comment by dcooney** [comment]
@landwire @CodingSamurai I started prototyping this however Gutenberg errors out when a block is missing.
![image](https://user-images.githubusercontent.com/428624/161602247-8fc5c6a4-6660-42c7-824a-d703b1d1ea5e.png)

The use case here is:
- Admin adds a block to a page that is only available to `admin` users.
- Editor logs in but doesn't have access to that block.
- Gutenberg throws an error for the editor.

They are still able to save the page but the experience here is a show stopper for this.

Any other ideas?

**Comment by farukgaric** [comment]
Hello @dcooney 

Nice plugin! I tried to limit settings only to Editor role by adding:
`if ( !current_user_can( 'editor' ) ) {
			return;
		}`
At the beginning this function:
https://github.com/dcooney/block-manager/blob/main/block-manager.php#L84

I believe it's good temporary solution. Of course, role should be retrieved from the settings page where we need select box with all roles.

My five cents! Keep doing great job!

Best
Faruk

**Comment by dcooney** [comment]
@farukgaric Missed your reply. My bad... there is a filter `gutenberg_block_manager_user_role` that can switch the access role for this plugin.
